building/upload-debs/rebuild.sh

```diff
@@ -1,7 +1,25 @@
 #!/bin/bash
 set -e -u
 
+if [ -z "${APT_REPOSITORY:-}" ]
+then
+	echo "Need to specify apt repository:"
+	echo "\$ export APT_REPOSITORY=<username>/<branch>"
+	exit 1
+fi
+
+if [ -z "${APT_SIGNING_KEY:-}" ]
+then
+	echo "Need to specify apt signing key:"
+	echo "\$ export APT_SIGNING_KEY=<key identifier>"
+	exit 1
+fi
+
+sed "s;\\\${APT_SIGNING_KEY};${APT_SIGNING_KEY};g" < conf/distributions.in > conf/distributions
+
 cd "$(dirname "$0")"
-reprepro -Vb . includedeb homeworld ../build-debs/binaries/homeworld-*.deb
-rsync -av --progress --delete-delay ./dists ./pool /mit/hyades/debian/
+gpg --import default-repo-signing-key.gpg
```

Comment:
I'm wary about automatically changing the state of the user's GPG keyring. Can you comment on whether this is necessary?

Reply:
I'm also uncomfortable with this part. However, reprepro doesn't seem to have a mechanism for using a key file directly. Looking at the man pages, though, it looks like I can specify a temporary GnuPG home directory for reprepro to use; I'll try doing that.

Reply:
Both of those work. Using a temporary GnuPG directory is good. Verifying that the key is in the keyring and telling the developer to import it if not is also good.